### ISSUE TITLE
Restructuring Repository

### ISSUE THREAD

**Issue by KurtJacobson** [KurtJacobson/hazzy#15]
@TurBoss My idea is to have multiple "modules" that are then embedded in the main GUI to make the complete interface. I experimented with this on the filechooser by developing it as a standalone application and then moving some of the files into the hazzy directory and embedding it into the main GUI. Overall this worked well but now I can't really go back to working on it separately again since I would have to manual sync all the changes.

So I am thinking I should rearrange the whole repository like this:

* LICENSE
* README.md
* TODO.md
* sim.hazzy
  * config.ini
  * config.ini 
  * ...
* hazzy
   * images
       * hazzy.glade 
       * image1.png
       * image2.png
       * ....
   * hazzy.py
   * preferances.py
   * ...
* plugins (or some name like that)
  * filechooser
      * filechooser.glade
      * filechooser.py
      * standalone.py
      * bookmarks.py
      * fileutilities.py
      * background.png
   * tooledit
       * tooledit.glade
       * tooledit.py
   * crazy_custom_display
     * crazy.py
     * crazy.glade

Basically, move the file chooser and other similar code that is essentially stand alone into sub folders.

Does that make any since, and it is possible to import modules form within other folders like this?
I have no clue what I am doing, so I understand if this is not possible or does not make any sense.

Thanks you again for all your help. I am learning a lot from you!
Kurt

**Comment by KurtJacobson** [comment]
OK I made a branch and rearranged my hazzy directory like above. Then to hazzy.py I added:
```
PLUGINDIR = os.path.realpath(os.path.join(HAZZYDIR, '..', 'plugins'))
sys.path.insert(2, PLUGINDIR)
```
And then imported the filechooser like this:
`from filechooser.filechooser import Filechooser`

It works!!! But is this a bad thing to do??

Thanks

**Comment by TurBoss** [comment]
Hello

    This seems good to me,

I would name "plugins" as "modules" and store glade files under a "ui" name folder or something under the root directory,  the ones for the modules are fine.

good?

to load modules from I don't really know any good method.

now I'm going to check out the branch to read a while.

was busy I Just finished my machine :-) 

if you would like to take a look I did a post under the linuxcnc forums : 
https://forum.linuxcnc.org/show-your-stuff/32728-fixed-bridge-machine-finished

**Comment by KurtJacobson** [comment]
Just pushed it to github as "restructure", but it does not look right. It did not delete the files in root?! I"ll have to work on that.

I like the name "modules" better than "plugins" too. Also good idea on having a separate "ui" folder. will make things more organized.

I think I came up with a better way to load modules. I'll make a modules.py that imports all the modules, and then import that in hazzy.py.

Off to look at your machine!!

**Comment by KurtJacobson** [comment]
Very nice machine!  I especially like the 7 segments DROs and your break out board!  Looks like it does accurate work. What material is that plaque made out of? I like it. 

Here is my machine writing LinuxCNC with a marker. That is always the very first test!!
https://youtu.be/SXHHHw69-NQ

**Comment by TurBoss** [comment]
its Krion
Wow your machine is fast :D 

How did the restructure go?

I'm goin to launch the simulator

**Comment by TurBoss** [comment]
I would move the modules under hazzy directory

**Comment by KurtJacobson** [comment]
The restructure went well. I am already finding it easer to work with the new file layout, so I think it was a good move.
I moved the modules under hazzy as you suggested, which makes importing them much easier.

**Comment by TurBoss** [comment]
I'm glad to be helpful

I'm in read only mode until you finish :)

**Comment by TurBoss** [comment]
I'm getting this error : 
```
The hazzy directory is: /home/turboss/Sources/hazzy/hazzy
Traceback (most recent call last):
  File "/home/turboss/Sources/hazzy/hazzy/hazzy.py", line 74, in <module>
    from dialogs.dialogs import Dialogs
```

imports should look like this:

```
from module.modulename import stuff
```

I'm taking a look

**Comment by TurBoss** [comment]
```

# Import modules
from modules.touchpads.keyboard import Keyboard
from modules.touchpads.touchpad import Touchpad
from modules.filechooser.filechooser import Filechooser
from modules.dialogs.dialogs import Dialogs
```

this fixed my problem

**Comment by KurtJacobson** [comment]
OK. I think that is best. When I had the modules in the root directory I could not import them this way, but now I can!! I though it would still have worked though as I added the modules directory to the path, hummm

**Comment by KurtJacobson** [comment]
Oh, I did not realize referencing an issue in a commit would close it. That is convenient!
